ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -872,21 +883,33 @@ private static boolean isInnerNode(final PlanNode node) {
     throw new IllegalStateException("Unknown node type: " + node.getClass().getName());
   }
 
-  private JoinKey buildForeignJoinKey(final Join join, final ColumnReferenceExp columnRef) {
+  private JoinKey buildForeignJoinKey(final Join join,
+                                      final Expression foreignKeyExpression) {
     final AliasedDataSource leftSource = join.getInfo().getLeftSource();
     final SourceName alias = leftSource.getAlias();
     final List<QualifiedColumnReferenceExp> leftSourceKeys =
         leftSource.getDataSource().getSchema().key().stream()
             .map(c -> new QualifiedColumnReferenceExp(alias, c.name()))
             .collect(Collectors.toList());
 
-    final ColumnName foreignKeyColumnName = columnRef.maybeQualifier().isPresent()
-        ? ColumnNames.generatedJoinColumnAlias(
-            columnRef.maybeQualifier().get(),
-            columnRef.getColumnName())
-        : columnRef.getColumnName();
+    final VisitParentExpressionVisitor<Optional<Expression>, Context<Void>> aliasRewritter =
+        new VisitParentExpressionVisitor<Optional<Expression>, Context<Void>>(Optional.empty()) {
+          @Override
+          public Optional<Expression> visitQualifiedColumnReference(
+              final QualifiedColumnReferenceExp node,
+              final Context<Void> ctx
+          ) {
+            return Optional.of(new UnqualifiedColumnReferenceExp(
+                ColumnNames.generatedJoinColumnAlias(node.getQualifier(), node.getColumnName())
```

Comment:
I'm not following the logic with qualifiers here. The old code only sometimes generated the join column alias (if a qualifier is present) but this new code seems to always generate the join column alias (before rewriting the alias to be unqualified). Is this a change in behavior?

Reply:
I think it not a behavior change, because we use the visiter pattern and this method is only executed for qualified column references (its name is `visitQualifiedColumnReference`). Thus unqualified column references should be untouched, as before.